homeassistant/components/tplink_omada/binary_sensor.py

```diff
@@ -31,26 +31,23 @@ async def async_setup_entry(
 ) -> None:
     """Set up binary sensors."""
     controller: OmadaSiteController = hass.data[DOMAIN][config_entry.entry_id]
-    omada_client = controller.omada_client
 
     gateway_coordinator = await controller.get_gateway_coordinator()
     if not gateway_coordinator:
         return
 
-    gateway = await omada_client.get_gateway(gateway_coordinator.mac)
-
-    async_add_entities(
-        get_gateway_port_status_sensors(gateway, hass, gateway_coordinator)
-    )
-
-    await gateway_coordinator.async_request_refresh()
+    for gateway in gateway_coordinator.data.values():
+        async_add_entities(
```

Comment:
Please collect the entities first and make a single call to add entities.

Reply:
Done